news-bundle/src/Resources/contao/dca/tl_news.php

```diff
@@ -62,7 +62,7 @@
 			(
 				'id' => 'primary',
 				'alias' => 'index',
-				'pid,start,stop,published' => 'index'
+				'pid,published,featured,start,stop' => 'index'
```

Comment:
I am pretty happy with the changes now, except for this one. Take the following query:

```sql
SELECT
    tl_news.*
FROM
    tl_news
WHERE
    tl_news.date>=0
    AND tl_news.date<=4294967295
    AND tl_news.pid IN(1)
    AND tl_news.published='1'
    AND (tl_news.start='' OR tl_news.start<='1599210660')
    AND (tl_news.stop='' OR tl_news.stop>'1599210720')
ORDER BY
    tl_news.date
```

With the `pid,published,featured,start,stop` index, key_len is 8 and filtered is 18.36%. However, without the `featured` column, key_len is 50 and filtered is 42.85%. That is a massive improvement, isn't it?

Reply:
That makes sense, doesn't it? Without featured your queried columns aren't a prefix of the index.

Imo low cardinality columns (like our "boolean" columns `published` & co) will likely not improve the performance for most of these queries. In your example it might be better having a separate BTREE on date which can be merged.

Reply:
Ok, so the best solution I found is to have a composite index `pid,start,stop,published` and a regular index on the `featured` column. WDYT?

Reply:
Then you would need two indexes I think, `pid,published,start,stop` and `pid,published,featured,start,stop`

Reply:
That did not work. MySQL still used the `pid,published,start,stop` index even though the query contained the `featured` column. But it worked with a separate index on the `featured` column.

Reply:
Probably the reason for this: If `featured = '1'` is set for only a few rows (which should be the normal case), this will likely be a good candidate for the *first* index being used by the query planner. This will result in a small set to start with when merging with the next index. If it's part of a multi-column index (i.e. the last step) using the index isn't much faster (or faster at all) than a scan of the already small result set.